internal/cmd/stack/open_command.go

```diff
@@ -110,22 +110,16 @@ func getGitRepositorySubdir() (string, error) {
 
 	root := current
 	for {
-		if _, err := os.Stat(".git"); err == nil {
+		if _, err := os.Stat(filepath.Join(root, ".git")); err == nil {
 			break
 		} else if !os.IsNotExist(err) {
 			return "", fmt.Errorf("couldn't stat .git directory: %w", err)
 		}
+		root = filepath.Dir(root)
+	}
 
-		cwd, err := os.Getwd()
-		if err != nil {
-			return "", fmt.Errorf("couldn't get current working directory: %w", err)
-		}
-
-		parent := filepath.Dir(cwd)
-		if err := os.Chdir(parent); err != nil {
```

Comment:
This to me - looks like a bug - I don't think looking up the sub dir should change the current working directory - my implementation is a fix for this - but there still may be a better way of detecting this. This also may have some weird side effect if it is loaded as a git submodule.